azurerm/internal/services/loganalytics/log_analytics_clusters_resource.go

```diff
@@ -0,0 +1,397 @@
+
+			"location": azure.SchemaLocation(),
+
+			"identity": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(operationalinsights.SystemAssigned),
+							}, false),
+						},
+
+						"principal_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"tenant_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"key_vault_property": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"key_vault_uri": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							DiffSuppressFunc: suppress.LogAnalyticsClusterUrl,
+						},
+
+						"key_version": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			// Per the documentation cluster capacity must start at 1000 GB and can go above 3000 GB with an exception by Microsoft
+			// so I am not limiting the upperbound here by design
+			// https://docs.microsoft.com/en-us/azure/azure-monitor/platform/manage-cost-storage#log-analytics-dedicated-clusters
+			"size_gb": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1000,
+				ValidateFunc: validation.All(
+					validation.IntAtLeast(1000),
+					validation.IntDivisibleBy(100),
+				),
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+func resourceArmLogAnalyticsClusterCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.ClusterClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	existing, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for present of existing Log Analytics Cluster %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_log_analytics_cluster", *existing.ID)
+	}
+
+	keyVaultProps := expandArmLogAnalyticsClusterKeyVaultProperties(d.Get("key_vault_property").([]interface{}))
+
+	if d.IsNewResource() && keyVaultProps != nil {
+		return fmt.Errorf("the Log Analytics Cluster %q (Resource Group %q) must be successfully provisioned before it can be configured to support customer managed keys", name, resourceGroup)
+	}
```

Comment:
The `d.IsNewResource` test is only needed when the create and update funcs are combined.  In this instance, it's not necessary here. Since the CMK element cannot be applied on create, it should probably be omitted entirely, and return the `Update` func instead of `Read` below.

Reply:
Fixed.

Reply:
I have determined through testing that this check needs to exist. Why? Because you cannot create the access policy in the key vault granting access to the cluster before the cluster resource has first been successfully provisioned. If you skip straight to the update function from the create function to apply the `key_vault_properties` the cluster will not have the correct permissions to access the key vault and will always receive the following error: 

```
Error: updating Log Analytics Cluster "acctest-LA-201112132727764751" (Resource Group "acctestRG-la-201112132727764751"): operationalinsights.ClustersClient#Update: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadRequest" Message="Failed to get access to encryption key https://vault201112132727764751.vault.azure.net:443/keys/key-98scj/***** due to the next error: Forbidden"
```

Is there a particular why you would prefer me to implement this check differently than I had originally implemented it?